examples/tinywl-new/wallpaperimage.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2024 UnionTech Software Technology Co., Ltd.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#include "wallpaperprovider.h"
+#include "wallpaperimage.h"
+#include "helper.h"
+#include "output.h"
+#include "woutput.h"
+#include "workspace.h"
+
+#include <QDir>
+
+WallpaperImage::WallpaperImage(QQuickItem *parent)
+    : QQuickImage(parent)
+{
+    auto provider = Helper::instance()->qmlEngine()->wallpaperImageProvider();
+    connect(provider, &WallpaperImageProvider::wallpaperTextureUpdate, this, &WallpaperImage::updateWallpaperTexture);
+
+    auto workspace = Helper::instance()->workspace();
+    connect(workspace, &Workspace::currentChanged, this, &WallpaperImage::updateSource);
```

Comment:
No, don't care current workspace here, the current workspace set by qml, for an example:

````
WallpaperImage {
    ...
    workspace: Helper.workspace.current
}
````